update gloo install guide for knative serving 0.8.0 (#1691)
knative/docs: docs/install/Knative-with-Gloo.md

---
title: "Knative Install using Gloo on a Kubernetes Cluster"
linkTitle: "Gloo"
weight: 10
type: "docs"
---

Learn how to deploy Gloo and Knative to your Kubernetes cluster using the Gloo command line tool `glooctl`.
 

[Gloo](https://gloo.solo.io) is a popular open-source Envoy control plane and API gateway built for Kubernetes (and other platforms). 

> Gloo provides a complete gateway replacement for Istio and supports the full Knative Ingress spec. Choose Gloo as a lightweight option that requires less resource usage and operational overhead compared to Istio.

Reviewer: ```suggestion
Gloo provides a complete gateway replacement for Istio and supports the full Knative Ingress spec. Choose Gloo if you don't require a service mesh in your cluster and want a lightweight alternative that requires less resource usage and operational overhead.
```

---
title: "Knative Install using Gloo on a Kubernetes Cluster"
linkTitle: "Knative with Gloo"
weight: 10
type: "docs"
---

[Gloo](https://gloo.solo.io) is a popular open-source Envoy control plane and API gateway built for Kubernetes (and other platforms). 

> Gloo can function as a complete replacement for Istio, supporting the full Knative Ingress spec. Using Knative with Gloo requires less resource usage and operational overhead than Istio.

Author: technically Gloo does not provide a `service mesh` replacement for Istio but a `gateway` replacement. I'll push that change

already created a Kubernetes cluster which you're comfortable installing _alpha_
software on.

> This guide assumes you are using bash in a Mac or Linux environment; some commands will need to be adjusted for use in a Windows environment.

Reviewer: Do you know which commands "will need to be adjusted"?

> 

Reviewer: @tcnghia  @ZhiminXiang is this correct? Are there other features that should be mentioned here?
```suggestion

- Some Knative features might not include support for Gloo. For example, the [automatic TLS certificate provisioning](../serving/using-auto-tls.md) is currently only supported in Istio.
```

Author: this is incorrect, Gloo supports the Automatic TLS feature (since it's included in the `Ingress` spec)

> ## Installing Glooctl

Reviewer: if not with code highlighting, this should be lower case (sentence style capitalization for section titles)

             env:
               - name: TARGET
                 value: Go Sample v1
   ```

   ```
   kubectl apply -f helloworld-go.yaml
   ```

2. Send a request

   **Knative Services** are exposed via the *Host* header assigned by Knative. By
   default, Knative will use the header `Host`:
   `{service-name}.{namespace}.example.com`. You can discover the appropriate *Host* header by checking the URL Knative has assigned to the `helloworld-go` service created above.

   ```
   $ kubectl get ksvc helloworld-go -n default  --output=custom-columns=NAME:.metadata.name,URL:.status.url
   ```
  
   ```
   NAME            URL
   helloworld-go   http://helloworld-go.default.example.com
   ```

   Gloo will use the `Host` header to route requests to the correct
   service. You can send a request to the `helloworld-go` service with curl
   using the `Host` and `$GATEWAY_URL` from above:

   ```
   $ curl -H "Host: helloworld-go.default.example.com" $(glooctl proxy url --name knative-external-proxy)
   ```
  
   ```
   Hello Go Sample v1!
   ```

Congratulations! You have successfully installed Knative with Gloo to manage and route to serverless applications!

> ## What's next

Reviewer: Please move this section to the bottom at the very end

Now that your cluster has Knative installed, you can see what Knative has to
offer.

To deploy your first app with Knative, follow the step-by-step
[Getting Started with Knative App Deployment](./getting-started-knative-app.md)
guide.

To get started with Knative Eventing, pick one of the
[Eventing Samples](../eventing/samples/) to walk through.

> Note that when following other Knative tutorials, you'll need to connect to the Gloo Gateway rather than the Istio Gateway when the tutorials prompts doing so.

> To get the URL of the Gloo Gateway, run

Reviewer: Move this above "Whats next"

```suggestion

## Connecting to the Gloo Gateway

Use the Gloo Gateway IP address and port number to connect to other services in your Knative cluster.

To get the URL of the Gloo Gateway, run
```

   `kubectl`

   ```yaml
   apiVersion: serving.knative.dev/v1alpha1
   kind: Service
   metadata:
     name: helloworld-go
     namespace: default
   spec:
     template:
       spec:
         containers:
           - image: gcr.io/knative-samples/helloworld-go
             env:
               - name: TARGET
                 value: Go Sample v1
   ```

   ```
   kubectl apply -f helloworld-go.yaml
   ```

2. Send a request

   **Knative Services** are exposed via the *Host* header assigned by Knative. By
   default, Knative will use the header `Host`:
   `{service-name}.{namespace}.example.com`. You can discover the appropriate *Host* header by checking the URL Knative has assigned to the `helloworld-go` service created above.

   ```
   $ kubectl get ksvc helloworld-go -n default  --output=custom-columns=NAME:.metadata.name,URL:.status.url
   ```
  
   ```
   NAME            URL
   helloworld-go   http://helloworld-go.default.example.com
   ```

   Gloo uses the `Host` header to route requests to the correct
   service. You can send a request to the `helloworld-go` service with curl
   using the `Host` and `$GATEWAY_URL` from above:

   ```

>    $ curl -H "Host: helloworld-go.default.example.com" $(glooctl proxy url --name knative-external-proxy)

Reviewer: ```suggestion
   curl -H "Host: helloworld-go.default.example.com" $(glooctl proxy url --name knative-external-proxy)
```
